code/modules/industrial_lift/elevator_button.dm

```diff
@@ -0,0 +1,164 @@
+		return
+
+	lift_weakref = WEAKREF(lift)
+
+// Emagging elevator buttons will disable safeties
+/obj/item/assembly/control/elevator/emag_act(mob/user, obj/item/card/emag/emag_card)
+	if(obj_flags & EMAGGED)
+		return
+
+	var/datum/lift_master/lift = lift_weakref?.resolve()
+	if(!lift)
+		return
+
+	for(var/obj/structure/industrial_lift/lift_platform as anything in lift.lift_platforms)
+		lift_platform.violent_landing = TRUE
+		lift_platform.warns_on_down_movement = FALSE
+		lift_platform.elevator_vertical_speed = initial(lift_platform.elevator_vertical_speed) * 0.5
+
+	// Note that we can either be emagged by having the button we are inside swiped,
+	// or by someone emagging the assembly directly after removing it (to be cheeky)
+	var/atom/balloon_alert_loc = get(src, /obj/machinery/button) || src
+	balloon_alert_loc.balloon_alert(user, "safeties overridden")
+	obj_flags |= EMAGGED
+	return TRUE
+
+// Multitooling emagged elevator buttons will fix the safeties
+/obj/item/assembly/control/elevator/multitool_act(mob/living/user)
+	if(!(obj_flags & EMAGGED))
+		return ..()
+
+	var/datum/lift_master/lift = lift_weakref?.resolve()
+	if(!lift)
+		return ..()
+
+	for(var/obj/structure/industrial_lift/lift_platform as anything in lift.lift_platforms)
+		lift_platform.violent_landing = initial(lift_platform.violent_landing)
+		lift_platform.warns_on_down_movement = initial(lift_platform.warns_on_down_movement)
+		lift_platform.elevator_vertical_speed = initial(lift_platform.elevator_vertical_speed)
+
+	// We can only be multitooled directly so just throw up the balloon alert
+	balloon_alert(user, "safeties reset")
+	obj_flags &= ~EMAGGED
+	return TRUE
+
+/obj/item/assembly/control/elevator/activate(mob/activator)
+	if(cooldown)
+		return
+
+	cooldown = TRUE
+	// Actually try to call the elevator - this sleeps.
+	// If we failed to call it, play a buzz sound.
+	if(!call_elevator(activator))
+		playsound(loc, 'sound/machines/buzz-two.ogg', 50, TRUE)
+
+	// Finally, give people a chance to get off after it's done before going back off cooldown
+	addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 2 SECONDS)
+
+/// Actually calls the elevator.
+/// Returns FALSE if we failed to setup the move.
+/// Returns TRUE if the move setup was a success, EVEN IF the move itself fails afterwards
+/obj/item/assembly/control/elevator/proc/call_elevator(mob/activator)
+	// We can't call an elevator that doesn't exist
+	var/datum/lift_master/lift = lift_weakref?.resolve()
+	if(!lift)
+		loc.balloon_alert(activator, "no elevator connected!")
+		return FALSE
+
+	// We can't call an elevator that's moving. You may say "you totally can do that", but that's not modelled
+	if(lift.controls_locked == LIFT_PLATFORM_LOCKED)
+		loc.balloon_alert(activator, "elevator is moving!")
+		return FALSE
+
+	// We can't call an elevator if it's already at this destination
+	var/obj/structure/industrial_lift/prime_lift = lift.lift_platforms[1]
+	if(prime_lift.z == loc.z)
```

Comment:
this is also potentially ambiguous for multiz lifts but there doesnt exist a solution for this presently